devito/interfaces.py

```diff
@@ -127,7 +132,10 @@ def set_initializer(self, lambda_initializer):
         self.initializer = lambda_initializer
 
     def _allocate_memory(self):
-        self._data = aligned(np.zeros(self.shape, self.dtype, order='C'), alignment=64)
+        if self.memmap:
+            self._data = aligned(np.memmap(filename=(self.f), dtype=self.dtype, mode='w+', shape=tuple(self.shape), order='C'), alignment=64)
```

Comment:
If I understand the implementation of np.memmap correctly, the call to aligned here will only waste memory and compute time without actually achieving any alignment since the buffer is allocated by python mmap and is not reallocated by numpy. Correct me if I am wrong.